File: py/Program/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    __jenis = ""
    __tangki = 0

    def __init__(self, jenis = "", tangki = 0, plat="", merk="", tahunProduksi=0, warna="", pemilik=""):
        super().__init__(plat, merk, tahunProduksi, warna, pemilik)
        self.__jenis = jenis
        self.__tangki = tangki
    
    def setJenis(self, jenis):
        self.__jenis = jenis
    
    def getJenis(self):
        return self.__jenis
    
    def setTangki(self, tangki):
        self.__tangki = tangki
    
    def getTangki(self):
        return self.__tangki

File: cpp/Program/Motorcycle.cpp
#pragma once
#include <bits/stdc++.h>
#include "Vehicle.cpp"

class Motorcycle : public Vehicle{
    private:
        string jenis;
        int tangki;
    
    public:
        Motorcycle(){
            
        }

        Motorcycle(string jenis, int tangki, string plat, string merk, int tahunProduksi, string warna, string pemilik) : Vehicle(plat, merk, tahunProduksi, warna, pemilik){
            this->jenis = jenis;
            this->tangki = tangki;
        }

        void setJenis(string jenis){
            this->jenis = jenis;
        }

        string getJenis(){
            return this->jenis;
        }

        void setTangki(int tangki){
            this->tangki = tangki;
        }

        int getTangki(){
            return this->tangki;
        }

        ~Motorcycle(){
            
        }
};